### PR TITLE
floating_ip: refactor for idempotency

### PR DESCRIPTION
**Pull request by resmo**

### CODE REVIEW

File: changelogs/fragments/floating-idempotency.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+- floating_ip - Added an optional name parameter to gain idempotency. The parameter will be required with release of version 2.0.0. (https://github.com/cloudscale-ch/ansible-collection-cloudscale/pull/43/).
```
Comment (felixfontein) [inline changelogs/fragments/floating-idempotency.yml]: To follow the official guidelines more closely:
```suggestion
- floating_ip - added an optional name parameter to gain idempotency. The parameter will be required with release of version 2.0.0 (https://github.com/cloudscale-ch/ansible-collection-cloudscale/pull/43/).
```

File: plugins/modules/floating_ip.py [hunk 2]
```diff
@@ -15,30 +15,38 @@
...
 author:
   - Gaudenz Steinlin (@gaudenz)
   - Denis Krienbühl (@href)
+  - René Moser (@resmo)
 version_added: 1.0.0
 options:
+  network:
+    description:
+      - Floating IP address to change.
+      - One of C(network) or C(name) is required to identify the floating IP.
```
Comment (felixfontein) [inline plugins/modules/floating_ip.py]:
```suggestion
      - One of I(network) or I(name) is required to identify the floating IP.
```
`I(...)` is for option names, `C(...)` for values.

File: plugins/modules/floating_ip.py [hunk 3]
```diff
@@ -189,83 +221,68 @@
...
+        return super(AnsibleCloudscaleFloatingIp, self).query()
+
+    def create(self, resource):
+        # Fail when missing params for creation
+        self._module.fail_on_missing_params(['ip_version'])
+
+        # TODO: To be removed in 2.0.0
+        if not self._module.params['name']:
+            self._module.deprecate(
+                msg="Name is missing, this won't be supported in the future.",
```
Comment (felixfontein) [inline plugins/modules/floating_ip.py]:
```suggestion
                msg="Name is missing, this won't be supported in the future",
```
IIRC deprecate adds a period at the end anyway. If you include one here, you'll get a double period.

File: plugins/modules/floating_ip.py [hunk 4]
```diff
@@ -277,36 +294,21 @@ def main():
 
     module = AnsibleModule(
         argument_spec=argument_spec,
-        required_one_of=(('ip', 'ip_version'),),
+        # TODO: Enable for version 2.0.0
+        # required_one_of=(('network', 'name'),),
```
Comment (felixfontein) [inline plugins/modules/floating_ip.py:287]: Before you said that `name` will be required. Now you say one of `network` and `name` will be required. Please decide :)

Comment (resmo) [reply plugins/modules/floating_ip.py:287]: Good point, the doc should say, "name will be required for creating a new floating IP"  adjusting that.

**Pull request merged**